Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Utilities/HttpRequestUtility.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace Amazon.Lambda.TestTool.Utilities;
+
+/// <summary>
+/// Utility class for handling HTTP requests in the context of API Gateway emulation.
+/// </summary>
+public static class HttpRequestUtility
+{
+    /// <summary>
+    /// Determines whether the specified content type represents binary content.
+    /// </summary>
+    /// <param name="contentType">The content type to check.</param>
+    /// <returns>True if the content type represents binary content; otherwise, false.</returns>
+    public static bool IsBinaryContent(string? contentType)
+    {
+        if (string.IsNullOrEmpty(contentType))
+            return false;
+
+        return contentType.StartsWith("image/", StringComparison.OrdinalIgnoreCase) ||
+               contentType.StartsWith("audio/", StringComparison.OrdinalIgnoreCase) ||
+               contentType.StartsWith("video/", StringComparison.OrdinalIgnoreCase) ||
+               contentType.Equals("application/octet-stream", StringComparison.OrdinalIgnoreCase);
+    }
+
+    /// <summary>
+    /// Reads the body of the HTTP request as a string.
+    /// </summary>
+    /// <param name="request">The HTTP request.</param>
+    /// <returns>The body of the request as a string.</returns>
+    public static string ReadRequestBody(HttpRequest request)
+    {
+        using (var reader = new StreamReader(request.Body))
+        {
+            return reader.ReadToEnd();
+        }
+    }
+
+    /// <summary>
+    /// Extracts headers from the request, separating them into single-value and multi-value dictionaries.
+    /// </summary>
+    /// <param name="headers">The request headers.</param>
+    /// <returns>A tuple containing single-value and multi-value header dictionaries.</returns>
+    /// <example>
+    /// For headers:
+    /// Accept: text/html
+    /// Accept: application/xhtml+xml
+    /// X-Custom-Header: value1
+    /// 
+    /// The method will return:
+    /// singleValueHeaders: { "Accept": "application/xhtml+xml", "X-Custom-Header": "value1" }
+    /// multiValueHeaders: { "Accept": ["text/html", "application/xhtml+xml"], "X-Custom-Header": ["value1"] }
+    /// </example>
+    public static (IDictionary<string, string>, IDictionary<string, IList<string>>) ExtractHeaders(IHeaderDictionary headers)
+    {
+        var singleValueHeaders = new Dictionary<string, string>();
+        var multiValueHeaders = new Dictionary<string, IList<string>>();
+
+        foreach (var header in headers)
+        {
+            singleValueHeaders[header.Key] = header.Value.Last() ?? "";
```

Comment:
My memory for API Gateway is that in HTTP V2 where there is just a single headers collection the headers are comma delimited. But we need to verify that. I'm sure it isn't just get the last value.

Reply:
yeah you are right on this https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html 

```
Format 2.0 doesn't have multiValueHeaders or multiValueQueryStringParameters fields. Duplicate headers are combined with commas and included in the headers field. Duplicate query strings are combined with commas and included in the queryStringParameters field.
```

i think i got confused when reading the other docs for v1 https://aws.amazon.com/blogs/compute/support-for-multi-value-parameters-in-amazon-api-gateway/

```
Before this change, API Gateway used to retain only the last value and drop everything else for a multi-valued parameter. You can see the original behavior in the queryStringParameters parameter in the above input, where only the “fish” value is retained.
```

I will update accordingly

Reply:
ive updated in fecf91d to handle this